cmd/counters/src/lib.rs

```diff
@@ -88,11 +233,22 @@ struct CountersArgs {
     #[clap(long, short, conflicts_with = "list")]
     full: bool,
 
+    /// show IPC counters, grouped by IPC interface rather than by counter.
+    #[clap(long, short, conflicts_with = "list")]
+    ipc: bool,
+
     /// sort counters using the provided ordering.
     ///
     /// [default: `decl` if `--full` is set, `alpha` otherwise]
     #[clap(long, short, conflicts_with = "list", value_enum)]
     sort: Option<Order>,
+
+    /// when used with `--ipc`, show only IPC counters originating from tasks
+    /// whose name contain the given substring.
+    ///
+    /// multiple values may be provided to select more than one client task.
+    #[clap(long, short, conflicts_with = "list", requires = "ipc")]
+    client: Vec<String>,
```

Comment:
As there's now a flag that's only relevant when `--ipc` is passed, I wonder if it would be a good idea to change `--ipc` from a flag to an `ipc` subcommand, so that the `--client` flag can only be defined for that subcommand? On the other hand, I haven't seen many other nested subcommands in Humility --- do we generally avoid nesting subcommands by convention?

Reply:
Bryan tends not to use more than one level of subcommand; I tend to use subcommands and positional arguments in preference to flags in a lot of situations. Kind of depends on who wrote the command.

I'd enthusiastically support subcommanding this case.